index.coffee

```diff
@@ -159,11 +159,13 @@ class ProgressReporter
 				console.warn('Progress error:', err.message ? err)
 
 exports.DockerProgress = class DockerProgress
-	constructor: (dockerOpts) ->
+	constructor: (opts = {}) ->
 		if !(this instanceof DockerProgress)
-			return new DockerProgress(dockerOpts)
-
-		@docker = new Docker(dockerOpts)
+			return new DockerProgress(opts)
+		if opts.docker?
+			@docker = opts.docker
```

Comment:
@Page- it actually has to be a DockerToolbelt, so I've now made it more explicit in that sense (the opts field has to be `dockerToolbelt`. In the one that we create, since dockerToolbelt forces Promise to be the bluebird one, we're fine.